### PR TITLE
⭐️ use cnspec for policy execution

### PR DESCRIPTION
**Pull request by chris-rock**
- update to go 1.19
- remove dependency on external mondoo binary 
- switch to use inventory instead of config piping to external binary

*cty dependency downgrade*

Packer and Terraform use different versions of go cty. The problem is that packer does not work with the latest version of go-cty as documented here:

- [hashicorp/packer-plugin-sdk incompatible with zclconf/go-cty v1.11.0 · Issue #134 · zclconf/go-cty · GitHub](https://github.com/zclconf/go-cty/issues/134)
- [Upgrade github.com/zclconf/go-cty to v1.11.1 · Issue #12055 · hashicorp/packer · GitHub](https://github.com/hashicorp/packer/issues/12055)
- [Upgrade github.com/zclconf/go-cty to v1.11.1 · Issue #135 · hashicorp/packer-plugin-sdk · GitHub](https://github.com/hashicorp/packer-plugin-sdk/issues/135)

There are two issues, we cannot bundle both go-cty versions due to the fact that it is not labeled a major change. Out only option to get this working for now is to downgrade cty in cnquery and cnspec.

*TODOs*

- [x] downgrade cty in cnquery https://github.com/mondoohq/cnquery/pull/499
- [x] downgrade cty in cnspec https://github.com/mondoohq/cnspec/pull/201
- [x] update readme

### CODE REVIEW

File: provisioner/provisioner.go [hunk 1]
```diff
@@ -196,7 +190,7 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 }
 
 func (p *Provisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.Communicator, generatedData map[string]interface{}) error {
-	ui.Say("Running Mondoo packer provisioner (Version: " + version.Version + ", Build: " + version.Build + ")")
+	ui.Say("Running Mondoo cnspec packer provisioner (Version: " + version.Version + ", Build: " + version.Build + ")")
```
Comment (scottford-io) [inline provisioner/provisioner.go:193]:
```suggestion
	ui.Say("Running cnspec packer provisioner by Mondoo (Version: " + version.Version + ", Build: " + version.Build + ")")
```

**Review by scottford-io — COMMENTED** [review]
@chris-rock I have tested both unregistered and registered builds on Linux and Windows. Linux works fine but Windows scans always produce the error: 

```bash
    mondoo-windows2019-secure-base-20221125182801.amazon-ebs.windows2019: scan packer build
==> mondoo-windows2019-secure-base-20221125182801.amazon-ebs.windows2019: scan failed: failed to resolve multiple assets
==> mondoo-windows2019-secure-base-20221125182801.amazon-ebs.windows2019: failed to resolve multiple assets
==> mondoo-windows2019-secure-base-20221125182801.amazon-ebs.windows2019: Stopping the source instance...
```

I am happy to take a crack at the `README` if you are ok with me adding to this branch. Just let me know.

File: .github/actions/spelling/expect.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+tions
```
Comment (tas50) [inline .github/actions/spelling/expect.txt:1]: make sure to have an empty line so this file processes correctly

**Review by scottford-io — APPROVED** [review]

File: provisioner/provisioner.go [hunk 3]
```diff
@@ -444,107 +411,171 @@ func (p *Provisioner) executeMondoo(ctx context.Context, ui packer.Ui, comm pack
...
+
+		// configure stderr logger
+		logger.Set("debug")
 	}
 
-	// pass incognito to mondoo scan
-	conf.Incognito = p.config.Incognito
+	// base 64 config env setting has always precedence
+	viper.SetConfigType("yaml")
+	if value := os.Getenv("MONDOO_CONFIG_BASE64"); len(value) > 0 {
```
Comment (preslavgerchev) [inline provisioner/provisioner.go:445]: Going back to my first comment about `MondooConfigPath` - this looks like a place where we might want to use it?

File: provisioner/provisioner.go [hunk 4]
```diff
@@ -4,32 +4,40 @@ package provisioner
 //go:generate packer-sdc struct-markdown
 
 import (
-	"bufio"
 	"bytes"
 	"context"
+	"encoding/base64"
```
[the remark this answers is not in the document]
Comment (chris-rock) [reply provisioner/provisioner.go:9]: @preslavgerchev Not sure I understand. Can you help me understand this?

File: provisioner/provisioner.go [hunk 5]
```diff
@@ -332,107 +325,81 @@ func (p *Provisioner) ConfigSpec() hcldec.ObjectSpec {
...
+		assetConfig.Insecure = true // we do not check the hostkey for the packer build
+		cred := vault.NewPasswordCredential(p.buildInfo.User, p.buildInfo.Password)
+		assetConfig.Credentials = append(assetConfig.Credentials, cred)
 	} else if p.buildInfo.ConnType == "docker" {
-		connType = "docker"
-		endpoint = fmt.Sprintf("%s", p.buildInfo.ID)
-		ui.Say(endpoint)
+		ui.Message("detected packer container image build")
+		assetConfig.Backend = providers.ProviderType_DOCKER
+		assetConfig.Host = fmt.Sprintf("%s", p.buildInfo.ID)
```
[the remark this answers is not in the document]
Comment (chris-rock) [reply provisioner/provisioner.go:386]: Yes

File: provisioner/provisioner.go [hunk 6]
```diff
@@ -444,107 +411,171 @@ func (p *Provisioner) executeMondoo(ctx context.Context, ui packer.Ui, comm pack
...
-	if p.config.Sudo != nil && p.config.Sudo.Active {
-		ui.Say("activated sudo")
-		conf.Sudo = VulnOptsSudo{
-			Active: p.config.Sudo.Active,
-		}
+	debugLogBuffer := &bytes.Buffer{}
+	logger.SetWriter(debugLogBuffer)
+	if p.config.Debug {
+		data, _ := json.Marshal(scanJob)
+		ui.Message(fmt.Sprintf("mondoo configuration: %v", string(data)))
```
[the remark this answers is not in the document]
Comment (chris-rock) [reply provisioner/provisioner.go:437]: Great feedback

File: provisioner/provisioner.go (hunk 3 above)
Comment (chris-rock) [reply provisioner/provisioner.go:445]: I am still not understanding what this means

File: provisioner/provisioner.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (preslavgerchev) [reply provisioner/provisioner.go:9]: Sorry, I should've been more clear with this (also cant comment on the actual variable as it's not in the diff) - the config (see `provisioner.go`)has a `MondooConfigPath` property which lets users specify where to search for a config file. We seem to not be using this anymore and only rely on the `MONDOO_CONFIG_PATH` and `MONDOO_CONFIG_BASE64` env vars. I believe we should also take this property into account and if provided - load the config from there

[the remark this answers is not in the document]
Comment (chris-rock) [reply provisioner/provisioner.go:9]: That is a great catch @preslavgerchev, I totally forgot about the config. I added that to the supported loading.

**Review by preslavgerchev — APPROVED** [review]

**Comment by preslavgerchev** [comment]
lgtm!

**Pull request merged**